GameServerLib/Content/Navigation/NavigationGrid.cs

```diff
@@ -0,0 +1,626 @@
+        }
+
+        public float CastRay(Vector2 origin, Vector2 destination, bool inverseRay = false)
+        {
+            return (float)Math.Sqrt(CastRaySqr(origin, destination, inverseRay));
+        }
+        public float CastRaySqr(Vector2 origin, Vector2 destination, bool checkWalkable = false)
+        {
+            float x1 = origin.X;
+            float y1 = origin.Y;
+            float x2 = destination.X;
+            float y2 = destination.Y;
+
+            if (x1 < 0 || y1 < 0 || x1 >= this.MapWidth || y1 >= this.MapHeight)
+            {
+                return 0.0f;
+            }
+
+            float distx = x2 - x1;
+            float disty = y2 - y1;
+            float greatestdist = Math.Abs(distx);
+            if (Math.Abs(disty) > greatestdist)
+            {
+                greatestdist = Math.Abs(disty);
+            }
+
+            int il = (int)greatestdist;
+            float dx = distx / greatestdist;
+            float dy = disty / greatestdist;
+            int i;
+            for (i = 0; i < il; i++)
+            {
+                //TODO: Implement bush logic (preferably near here)
+                //TODO: Implement methods for NavGrids without SEE_THROUGH flags
+                if (IsWalkable(x1, y1) == checkWalkable && IsSeeThrough(x1, y1) == checkWalkable)
+                {
+                    break;
+                }
+
+                // if checkWalkable == true, stop incrementing when (x1, x2) is a see-able position
+                // if checkWalkable == false, stop incrementing when (x1, x2) is a non-see-able position
+                x1 += dx;
+                y1 += dy;
+            }
+
+            if (i == il || (x1 == origin.X && y1 == origin.Y))
+            {
+                return (new Vector2(x2, y2) - origin).SqrLength();
+            }
+
+            return (new Vector2(x1, y1) - origin).SqrLength();
+        }
+        public float CastInfiniteRaySqr(Vector2 origin, Vector2 direction, bool inverseRay = false)
+        {
+            return CastRaySqr(origin, origin + direction * 1024, inverseRay);
+        }
+        public float CastInfiniteRay(Vector2 origin, Vector2 direction, bool inverseRay = false)
+        {
+            return (float)Math.Sqrt(CastInfiniteRaySqr(origin, direction, inverseRay));
+        }
+
+        public bool IsAnythingBetween(Vector2 a, Vector2 b)
+        {
+            return CastRaySqr(a, b) < (b - a).SqrLength();
+        }
+        public bool IsAnythingBetween(IGameObject a, IGameObject b)
+        {
+            if (a is IObjBuilding)
+            {
+                double rayDist = Math.Sqrt(CastRaySqr(b.GetPosition(), a.GetPosition()));
+                rayDist += a.CollisionRadius;
+                return (rayDist * rayDist) < (b.GetPosition() - a.GetPosition()).SqrLength();
+            }
+            else
+            {
+                return CastRaySqr(a.GetPosition(), b.GetPosition()) < (b.GetPosition() - a.GetPosition()).SqrLength();
```

Comment:
Same problem.

Reply:
What's the problem here ? Everything looks correct to me

Reply:
> 
> 
> What's the problem here ? Everything looks correct to me

The problem is that you changed the current code: https://github.com/LeagueSandbox/GameServer/blob/8e9ab428855c22316af9978678d68f6c8d91b15e/GameServerLib/Content/NavGrid.cs#L553

Reply:
Fixed 👌